src/main/java/com/powsybl/caseserver/service/S3CaseService.java

```diff
@@ -0,0 +1,558 @@
+        }
+
+        String key = uuidToPrefixKey(caseUuid) + fileName;
+        // For compressed cases, we append the compression extension to the case name as only the compressed file is stored in S3.
+        // i.e. : Assuming test.xml.gz is stored in S3. When you request datasourceExists(randomUUID, "test.xml"), you ask to S3 API ("test.xml" + ".gz") exists ? => true
+        if (isCompressedCaseFile(getCaseName(caseUuid))) {
+            key = "." + getCompressionFormat(caseUuid);
+        }
+
+        HeadObjectRequest headObjectRequest = HeadObjectRequest.builder()
+                .bucket(bucketName)
+                .key(key)
+                .build();
+        try {
+            s3Client.headObject(headObjectRequest);
+            return Boolean.TRUE;
+        } catch (NoSuchKeyException e) {
+            return Boolean.FALSE;
+        }
+    }
+
+    static boolean isCompressedCaseFile(String caseName) {
+        return COMPRESSION_FORMATS.stream().anyMatch(cf -> caseName.endsWith("." + cf));
+    }
+
+    static boolean isArchivedCaseFile(String caseName) {
+        return ARCHIVE_FORMATS.stream().anyMatch(cf -> caseName.endsWith("." + cf));
+    }
+
+    public Set<String> listName(UUID caseUuid, String regex) {
+        List<S3Object> s3Objects = getCaseFileSummaries(caseUuid);
+        List<String> fileNames = s3Objects.stream().map(obj -> Paths.get(obj.key()).toString().replace(CASES_PREFIX + caseUuid.toString() + DELIMITER, "")).toList();
+        if (Objects.nonNull(getCompressionFormat(caseUuid)) && getCompressionFormat(caseUuid).equals("zip")) {
+            fileNames = fileNames.stream().filter(name -> !name.equals(getOriginalFilename(caseUuid))).toList();
+        } else if (isCompressedCaseFile(fileNames.get(0))) {
+            fileNames = List.of(fileNames.get(0).replace("." + getCompressionFormat(caseUuid), ""));
+        }
+        return fileNames.stream().filter(n -> n.matches(regex)).collect(Collectors.toSet());
+    }
+
+    @Override
+    public UUID importCase(MultipartFile mpf, boolean withExpiration, boolean withIndexation) {
+        UUID caseUuid = UUID.randomUUID();
+
+        String caseName = mpf.getOriginalFilename();
+        validateCaseName(caseName);
+
+        String format = withTempCopy(caseUuid, caseName, mpf::transferTo, this::getFormat);
+        String compressionFormat = null;
+
+        try (InputStream inputStream = mpf.getInputStream()) {
+            String key = uuidAndFilenameToKey(caseUuid, caseName);
+
+            Map<String, String> caseMetadata = new HashMap<>();
+            caseMetadata.put(FORMAT_HEADER_KEY, format);
+
+            if (isArchivedCaseFile(caseName)) {
+                importZipContent(mpf.getInputStream(), caseUuid);
+                compressionFormat = FileNameUtils.getExtension(Paths.get(caseName));
+            } else if (isCompressedCaseFile(caseName)) {
+                compressionFormat = FileNameUtils.getExtension(Paths.get(caseName));
```

Comment:
caseName can be null
Path caseNameFromPath = Paths.get(Objects.requireNonNull(caseName)); ?

Reply:
Done at caseName initialization